### ISSUE TITLE
std::common_type specializations vs out-of-the box support

### ISSUE THREAD

**Issue by apolukhin** [ZaMaZaN4iK/numeric-papers-in-latex#13]
[LEWGI review](https://github.com/cplusplus/papers/issues/638#issuecomment-552046973) noted that we may avoid std::common_type specializations for wide_integers if we can make ternary operator for wide_integers behave as intended.

We need some experiments and either an in detail description of why ternary operator fails or a paper that addresses the problem and removes the std::common_type specializations

**Comment by apolukhin** [comment]
@nickaleks volunteered to take care of the issue.

**Comment by JohelEGP** [comment]
I didn't know this was possible. Here's a sample implementation (https://godbolt.org/z/hfv3Yh):
```C++
#include <type_traits>
template <int B> struct Int {
    Int() = default;
    template <int B2> Int(Int<B2>) requires (B2 < B) { }
};
static_assert(std::is_same_v<Int<2>,
                             std::common_type_t<Int<2>, Int<1>>>);
```

**Comment by nickaleks** [comment]
> I didn't know this was possible. Here's a sample implementation (https://godbolt.org/z/hfv3Yh):
> 
> ```c++
> #include <type_traits>
> template <int B> struct Int {
>     Int() = default;
>     template <int B2> Int(Int<B2>) requires (B2 < B) { }
> };
> static_assert(std::is_same_v<Int<2>,
>                              std::common_type_t<Int<2>, Int<1>>>);
> ```

Doesn't that prevent narrowing conversion?

**Comment by JohelEGP** [comment]
Yes. I guess `wide_integer`'s desire to be implicitly convertible to any other `wide_integer` as do their underlying types is at odds with `std::common_type` working without explicit specialization. I guess that's the answer: the design choice of acting like the underlying integer type requires explicit `std::common_type` specializations.
